design/src/rfcs/rfc0017_logging_sensitive.md

```diff
@@ -0,0 +1,376 @@
+        fut.instrument(span)
+    }
+}
+```
+
+As this path latency-sensitive, careful implementation is required to avoid excess allocations during redaction of sensitive data. Wrapping `Uri` and `HeaderMap` then providing a new `Display` implementation which skips over the sensitive data is preferable over allocating a new `String`/`HeaderMap` and mutating it.
+
+### Internal Guideline
+
+A guideline should be made available to internal smithy developers to outline the following:
+
+- The [HTTP bindings traits](#http-binding-traits) and why they are of concern
+- The [Debug implementation](https://github.com/awslabs/smithy-rs/pull/229) on structures
+- How to use the `Sensitive` struct and the `debug-logging` feature flag described in [Debug Logging](#debug-logging)
+
+### Public Guideline
+
+A guideline should be made available to customers to outline the following:
+
+- The [HTTP bindings traits](#http-binding-traits) and why they are of concern.
+- Warn against the two potential leaks described in [Scope and Guidelines](#scope-and-guidelines)
+  - Sensitive data leaking from third-party dependencies
+  - Sensitive data leaking from middleware applied to the `Router`
+- How to use the `Sensitive` struct and the `debug-logging` feature flag described in [Debug Logging](#debug-logging)
+
+## Alternative Proposals
+
+All of the following proposals are compatible with, and benefit from, [Debug Logging](#debug-logging), [Internal Guideline](#internal-guideline)/[External Guideline](#external-guideline) portions of the main proposal.
+
+The main proposal disallows the logging of potentially sensitive data in the runtime crates, instead opting for a dedicated code generated logging layer. In contrast, the following proposals all seek ways to accommodate logging of potentially sensitive data in the runtime crates.
+
+### Use Request Extensions
+
+Request extensions are mechanism to adjoin data to a Request as it passes through the middleware layers - the type map [http::Extensions](https://docs.rs/http/latest/http/struct.Extensions.html).
+
+These can be used to provide data to middleware interested in logging potentially sensitive data.
+
+```rust
+struct Sensitivity {
+    /* Data concerning which parts of the request are sensitive */
+}
+
+struct Middleware<S> {
+    inner: S
+}
+
+impl<B, S> Service<Request<B>> for Middleware<S> {
+    /* ... */
+
+    fn call(&mut self, request: Request<B>) -> Self::Future {
+        if let Some(sensitivity) = request.extensions().get::<Sensitivity>() {
+            if sensitivity.is_method_sensitive() {
+                debug!(method = %request.method());
+            }
+        }
+
+        /* ... */
+
+        self.inner.call(request)
+    }
+}
+```
+
+A middleware layer must be code generated (much in the same way as the logging layer) which is dedicated to inserting the `Sensitivity` struct into the extensions of each incoming request.
+
+```rust
+impl<B, S> Service<Request<B>> for SensitivityInserter<S>
```

Comment:
This also runs into the same problem as . We want `Request<B>` to get the extension as soon as routing has been done and we know which operation it's being passed to.